### PR TITLE
Advertise SELinux support

### PR DESCRIPTION
**Pull request by DemiMarie**
Also some cleanups.  See individual commit messages for details.

Marked as draft because this has received no testing (yet).

### CODE REVIEW

File: template_rpm/09_cleanup.sh [hunk 1]
```diff
@@ -49,4 +49,9 @@ fi
 
 truncate --no-create --size=0 "${INSTALL_DIR}"/var/log/dnf.*
 
+if [[ "$DIST_NAME" = 'fedora' ]] && [[ "$DIST_VER" -ge 37 ]] &&
+    rpm --root "$INSTALL_DIR" -qv qubes-core-agent-selinux; then
+    echo 'selinux=1' > /template.conf
```
Comment (marmarek) [inline template_rpm/09_cleanup.sh:54]: I'm pretty sure `/template.conf` is not correct here. See https://github.com/QubesOS/qubes-linux-template-builder/blob/main/qubeize_image#L117-L132 where it's looked for. Using `./template.conf` would work, as the "cleanup" script is called after that selection, but it isn't particularly nice.
The current template builder code covers only "static" template.conf (based on distribution version/flavor), not dynamically created. So, maybe add another case there that looks for example into `$(dirname "$IMG")` (or a new variable with this value)? And then generate template.conf in 04_install_qubes.sh.

File: template_rpm/09_cleanup.sh [hunk 2]
```diff
@@ -49,4 +49,9 @@ fi
 
 truncate --no-create --size=0 "${INSTALL_DIR}"/var/log/dnf.*
 
+if [[ "$DIST_NAME" = 'fedora' ]] && [[ "$DIST_VER" -ge 37 ]] &&
+    rpm --root "$INSTALL_DIR" -qv qubes-core-agent-selinux; then
```
Comment (marmarek) [inline template_rpm/09_cleanup.sh:53]: I'd recommend `chroot "$INSTALL_DIR" rpm ...` instead. You don't have guarantee that build host's rpm is compatible with rpmdb inside the template.

**Comment by marmarek** [comment]
Besides specific comments above, this needs also:
1. Installing selinux policy
2. relabeling fs at the very end
3. creating flag file for qubes scripts that the fs is already relabeled.

All of the above should be conditional on some template option/flavor, you can for example use `containsFlavor "selinux"` as the condition. And then build the template as "fc37+selinux" for example (or put that into "options" list if using builderv2).
Currently linux-template-builder is duplicated into builderv2. You can use whichever you like, I'll port that part (if any changes actually will be needed) to the other one when merging.

**Comment by DemiMarie** [comment]
> All of the above should be conditional on some template option/flavor, you can for example use `containsFlavor "selinux"` as the condition. And then build the template as "fc37+selinux" for example (or put that into "options" list if using builderv2).

What is the advantage of making this conditional?  The SELinux package is not very large and disabling it after the fact is easy.

**Comment by marmarek** [comment]
(I'm not sure how to make a comment to a whole commit in a PR, so I'm posting it here)

Please drop the google-chrome removing commit in its current shape. First of all, the claim in the commit message is false - packages seems to be signed with DSA key, not RSA. But then, if you read the comment you remove, you can see where you can get a signing key. The file under that URL contains actually two keys: DSA 1024 used to sign packages, and RSA 4096 used to sign repo metadata. So, a much better fix is to update the repo to actually use the latter. But please do that in a separate PR.

**Comment by marmarek** [comment]
> What is the advantage of making this conditional? The SELinux package is not very large and disabling it after the fact is easy.

First of all, adding it unconditionally will break all template builds for R4.1, as selinux policy is only in R4.2. But also, it will pull in extra dependencies on minimal template - which already is quite big as a "minimal" one.  And no, don't just add another check for qubes version - the correct place for this selection in builder config. This will avoid the need for any further changes here, for example if somebody would want centos template with selinux enabled (possibly after adjusting few other packages first).

File: template_rpm/packages_fedora.list [hunk 3]
```diff
@@ -26,6 +26,7 @@ man-db
 man-pages
 network-manager-applet
 rsync
+selinux-policy-targeted
```
Comment (marmarek) [inline template_rpm/packages_fedora.list:29]: This still needs to be conditional on the `selinux` build option. Which means you need to do that in some of the scripts, not the list file.

**Comment by marmarek** [comment]
```
+ unshare --mount -- chroot -- /builder/mnt /bin/sh -euc 'mount --bind -- / /mnt
        umask 0755
        mkdir -p -m 0700 -- /dev /var /run
        mkdir -p -m 1777 -- /tmp /var/tmp /dev/shm
        find /tmp /var/tmp /run /dev/shm -mindepth 1 -delete 
        : > /.qubes-relabeled
        rm -f /.autorelabel
        setfiles -r /mnt -- "/etc/selinux/$1/contexts/files/file_contexts"' sh targeted
usage:  setfiles [-diIDlmnpqvCEFWT] [-e excludedir] [-r alt_root_path] [-c policyfile] spec_file pathname...
usage:  setfiles [-diIDlmnpqvCEFWT] [-e excludedir] [-r alt_root_path] [-c policyfile] spec_file -f filename
usage:  setfiles -s [-diIDlmnpqvFWT] spec_file
```

**Comment by marmarek** [comment]
Ok, now relabeling works. There are two remaining issues:
1. When building with builderv2, the current directory is different, so template.conf modification doesn't work. I'll add variable with explicit directory for that and generally take care of this point.
2. Selinux remains in permissive mode. That's still better than not enabled, as one can actually enable it manually and have it working immediately (labels seems to be properly preserved), but it's more useful if it would be enforcing out of the box :)

File: template_rpm/09_cleanup.sh [hunk 4]
```diff
@@ -49,4 +51,17 @@ fi
...
+    sed -i -- 's/^SELINUX=\(disabled\|enforcing\)/SELINUX=permissive/' "$INSTALL_DIR/etc/selinux/config"
+    unshare --mount -- chroot -- "$INSTALL_DIR" /bin/sh -euc 'mount --bind -- / "$2"
+        umask 0755
+        mkdir -p -m 0700 -- /dev /var /run
+        mkdir -p -m 1777 -- /tmp /var/tmp /dev/shm
+        find /tmp /var/tmp /run /dev/shm -mindepth 1 -delete
+        : > /.qubes-relabeled
+        rm -f /.autorelabel
+        setfiles -r "$2" -- "/etc/selinux/$1/contexts/files/file_contexts" "$2"' sh targeted /mnt
+    echo 'selinux=1' > ./template.conf
```
Comment (marmarek) [inline template_rpm/09_cleanup.sh:64]: Ok, with the two PRs linked to https://github.com/QubesOS/qubes-issues/issues/7988, you can now do:
```
echo 'selinux=1' > "$TEMPLATE_CONF"
```

**Comment by DemiMarie** [comment]
PipelineRetryFailed

**Review by marmarek — APPROVED** [review]

**Pull request merged**